packages/wagmi/src/utils/defaultWagmiCoreConfig.ts

```diff
@@ -60,9 +60,10 @@ export function defaultWagmiConfig({
   if (enableCoinbase !== false) {
     connectors.push(
       coinbaseWallet({
+        version: '4',
         appName: metadata?.name ?? 'Unknown',
         appLogoUrl: metadata?.icons[0] ?? 'Unknown',
-        enableMobileWalletLink: true
+        preference: 'all'
```

Comment:
Since this `preference` doesn't tell what exactly it is, could we add a JSDoc for it and reference to their docs?
```suggestion
    /**
     * Determines which wallet options to display in Coinbase Wallet SDK.
     * @property preference
     *   - `all`: Show both smart wallet and EOA options.
     *   - `smartWalletOnly`: Show only smart wallet options.
     *   - `eoaOnly`: Show only EOA options.
     * @see https://www.smartwallet.dev/sdk/v3-to-v4-changes#parameters
     */
    preference: "all";

```